src/extension/extensionServer.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as em from "../common/extensionMessaging";
+import {FileSystem} from "../common/node/fileSystem";
+import {Packager} from "../common/packager";
+import {Log} from "../common/log";
+import * as Q from "q";
+import * as net from "net";
+import * as vscode from "vscode";
+
+
+export class ExtensionServer implements vscode.Disposable {
+    private outputChannel: any;
+    private serverInstance: net.Server = null;
+    private messageHandlerDictionary: { [id: number]: ((...argArray: any[]) => Q.Promise<any>) } = {};
+    private reactNativePackager: Packager;
+
+    public constructor(reactNativePackager: Packager) {
+        this.outputChannel = vscode.window.createOutputChannel("React-Native");
+        this.reactNativePackager = reactNativePackager;
+
+        /* register handlers for all messages */
+        this.messageHandlerDictionary[em.ExtensionMessage.START_PACKAGER] = this.startPackager;
+        this.messageHandlerDictionary[em.ExtensionMessage.STOP_PACKAGER] = this.stopPackager;
+        this.messageHandlerDictionary[em.ExtensionMessage.PREWARM_BUNDLE_CACHE] = this.prewarmBundleCache;
+    }
+
+    /**
+     * Starts the server.
+     */
+    public setup(): Q.Promise<void> {
+
+        let deferred = Q.defer<void>();
+
+        let launchCallback = (error: any) => {
+            Log.logMessage("Server started." + (error ? error : ""));
```

Comment:
We should call logError if you have an error. Should we print a more detailed message to the user if he gets an error here? He probably won't be able to use the debugging functionality.